lib/shared/src/chat/typewriter.ts

```diff
@@ -113,9 +118,16 @@ export class Typewriter implements IncrementalTextConsumer {
         }
         // Clean up the consumer, finished promise.
         if (this.upstreamClosed) {
-            this.consumer.close()
-        } else {
-            console.error('Typewriter stopped', error)
+            if (error) {
```

Comment:
```suggestion
            if (error) {
                console.error('Typewriter stopped', error)
                if (this.consumer.error) {
                    this.consumer.error(error)
                    return
                }
            }
            this.consumer.close()
```
would this do the same thing?